frontend/helpers/sepa.tsx

```diff
@@ -50,125 +35,191 @@ const generateSEPAHeader = (data) => ({
-        else if (nature === RENTAL_EXPENSES) rentalExpenseTransactions.push(historyData);
-        else currentExpenseTransations.push(historyData);
-        allTransactions.push({ fields: historyData });
-      }
-    }
-
-    return { rentTransactions, rentalExpenseTransactions, currentExpenseTransations, allTransactions };
-  } catch (e) {
-    addMessageAndThrow(e, 'error when formatting transaction data');
-  }
+const generateTransactions = (amounts, roommatesData, configData, transactionsHistoryData) => {
+  const transactionMonthCount = transactionsHistoryData.filter(h => dayjs().isSame(h.date, 'month')).length || 0;
+
+  const transactionsRent = generateTransactionsForOnePayment({
+    roommatesData,
+    amount: amounts.rent,
+    expenseLabel: configData.rentLabel,
+    transactionMonthCount,
+    creditorPrefix: configData.creditorPrefix
+  });
+  const transactionsRentalExpenses = generateTransactionsForOnePayment({
+    roommatesData,
+    amount: amounts.rentalExpenses,
+    expenseLabel: configData.rentalExpensesLabel,
+    transactionMonthCount: transactionMonthCount + transactionsRent.length,
+    creditorPrefix: configData.creditorPrefix
+  });
+  const transactionsCurrentExpenses = generateTransactionsForOnePayment({
+    roommatesData,
+    amount: amounts.currentExpenses,
+    expenseLabel: configData.currentExpensesLabel,
+    transactionMonthCount: transactionMonthCount + transactionsRent.length + transactionsRentalExpenses.length,
+    creditorPrefix: configData.creditorPrefix
+  });
+
+  return { transactionsRent, transactionsRentalExpenses, transactionsCurrentExpenses };
 };
 
 export const downloadSEPAXml = async (amounts) => {
   try {
     const configData = await getConfigData();
     const roommatesData = await getRoommatesData();
-    const formattedRoommatesData = roommatesData.map(rm => ({
-      _id: 'id',
-      number: 'REG1234567',
-      netInclTaxes: '80',
-      rum: rm.debitorRUM,
-      customerInfo: { payment: { bic: rm.debitorBIC, bankAccountOwner: rm.debitorName, iban: rm.debitorIBAN } }
-    }));
+    const transactionsHistoryData = await getTransactionsHistoryData();
+    const randomId = randomize('0', 21);
+
+    
+    const {
+      transactionsRent,
+      transactionsRentalExpenses,
+      transactionsCurrentExpenses
+    } = generateTransactions(amounts, roommatesData, configData, transactionsHistoryData);
+    const allTransactions = [...transactionsRent, ...transactionsRentalExpenses, ...transactionsCurrentExpenses];
+    
+    const rentTotalAmount = transactionsRent.reduce((acc, next) => acc + next.amount, 0);
+    const rentalExpensesTotalAmount = transactionsRentalExpenses.reduce((acc, next) => acc + next.amount, 0);
+    const currentExpensesTotalAmount = transactionsCurrentExpenses.reduce((acc, next) => acc + next.amount, 0);
+    
+    let rentPaymentInfo;
+    if (transactionsRent.length) {
+      rentPaymentInfo = generatePaymentInfo({
+        id: `MSG0LYER${randomId}R`,
```

Comment:
Comment as-tu choisi le format de ces ids ? Dans ta description tu dis "on conseille de se limiter a 30", qui est ce "on" ?

Reply:
https://www.febelfin.be/sites/default/files/2019-04/message_xml_pour_lordre_de_virement_-_version_3.3.pdf
page 20

Ca concerne MsgId,  et du coup comme on a uniformisé les formats, ca se repercute sur PmtInfId